### DISCUSSION TITLE
[Feature Request] premoderation support

### DISCUSSION THREAD

**Discussion by halittiryaki** [umputun/remark42#1236] [Ideas]
currently there is no way of premoderation. comments are published immediately and can be deleted by an admin later.
it should be possible to enable a premoderation mode, where a comment is only publicy visible after an admin user approves it.

**Comment by umputun** [comment]
moving to discussions as this not an actual issue

**Reply by paskal** [reply]
We already have a discussion about it: https://github.com/umputun/remark42/discussions/1019

**Comment by umputun** [comment]
There are two ways to implement it. One possibility is probably easy to do - if this pre-moderation flag is set we publish each new comment with a soft delete initially and latter on admin may "undelete" it. As far as I recall we already have API to do it. This solution can cause some unexpected complexity, for example, notification should be sent to the admin "channel" right away but postponed till approval before it is sent to users. Another problem with this solution - it is some kind of hack and `delete` status is not a proper substitution for the "enabled" (or "approved") flag

The proper solution is likely harder to implement, i.e. add `enabled` field. The backend change to support this new flag will require several changes in multiple places, for instance, need a way to get only those with enabled=false, change it to true, exclude comments from the regular (non-admin) view prior to approval, and so on. It will also need UI change, to approve/disapprove
